SAP/cloud-sdk-js
# chore: mail poc

This PR contains the following:
- ADR for supporting the mail feature 
- update `auth-flow` tests
- add a mail e2e test
- add a proxy e2e test

TODO:
- [x] double check package json scripts

Closes SAP/cloud-sdk-backlog#ISSUENUMBER.
